File: library/src/main/java/com/github/daneko/android/iab/model/SkuDetails.java
package com.github.daneko.android.iab.model;

import org.json.JSONObject;

import javax.annotation.Nonnull;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Value;

/**
 * Represents an in-app product's listing details.
 */
@Value
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@EqualsAndHashCode(exclude = {"json"})
public class SkuDetails {
    String sku;
    String type;
    String price;
    String title;
    String description;
    JSONObject json;

    public static SkuDetails create(@Nonnull final JSONObject jsonSkuDetails) {

        return new SkuDetails(
                jsonSkuDetails.optString("productId"),
                jsonSkuDetails.optString("type"),
                jsonSkuDetails.optString("price"),
                jsonSkuDetails.optString("title"),
                jsonSkuDetails.optString("description"),
                jsonSkuDetails
        );
    }

    /**
     * alias {@link #getSku()}
     */
    public String getProductId() {
        return sku;
    }

}

File: library/src/main/kotlin/com/github/daneko/android/iab/model/SkuDetails.kt
package com.github.daneko.android.iab.model

import org.json.JSONObject

/**
 * Represents an in-app product's listing details.
 */
data class SkuDetails private constructor(
        internal val sku: String,
        val type: String,
        val price: String,
        val title: String,
        val description: String,
        val json: JSONObject
) {
    /**
     * alias [.getSku]
     */
    fun getProductId() = sku

    companion object {
        @JvmStatic
        fun create(jsonSkuDetails: JSONObject): SkuDetails {
            return SkuDetails(
                    jsonSkuDetails.optString("productId"),
                    jsonSkuDetails.optString("type"),
                    jsonSkuDetails.optString("price"),
                    jsonSkuDetails.optString("title"),
                    jsonSkuDetails.optString("description"),
                    jsonSkuDetails)
        }
    }
}
